index.js

```diff
@@ -71,13 +71,14 @@ const getAllFindings = async (ECR, repository, tag) => {
       nextToken: NextMarker
     }).promise().catch(
       (err) => {
+        core.debug(`Error: ${err}`);
         if (err.code === 'ScanNotFoundException') { return null }
         throw err
       })
 
     return {
       marker: findings.nextToken,
-      results: findings.imageScanFindings.findings,
+      results: findings.imageScanFindings.findings || findings.imageScanFindings.enhancedFindings,
```

Comment:
Great stuff!

I see that you've put a friendly message for basic scanning results further into the call stack. I think instead we should [pull the error check up](https://matklad.github.io/2023/11/15/push-ifs-up-and-fors-down.html), and only pass in enhanced findings.

Something like:

```js
    // ...existing code
    }).promise().catch(
      (err) => {
        core.debug(`Error: ${err}`);
        if (err.code === 'ScanNotFoundException') { return null }
        throw err
      })

    // new code
    if (!findings.imageScanFindings.enhancedFindings) {
          throw new Error(`Basic scan not supported. Please enable enhanced scanning in ECR.`)
    }

    return {
      marker: findings.nextToken,
      results: findings.imageScanFindings.enhancedFindings,
// ...
```

Then you can simplify further down, like removing `isEnhancedScan` and inlining `getEnhancedScanFindings`.

Reply:
Let me know if the changes I made are what you're looking for. I moved the code to `getFindings` because we have 2 functions and `getFindings` is called first so that's where the `enhancedScan` check should happen. The `getAllFindings` is paginated which makes changing that one a little more challenging.